docs/diagrams/AddCommandSequenceDiagram.puml

```diff
@@ -0,0 +1,41 @@
+@startuml
+hide footbox
+
+actor user
+participant ":Ui" as Ui
+participant ":Parser" as Parser
+participant ":AddCommand" as AddCommand
+
+activate Ui
+user -> Ui : "add..."
+create Parser
+Ui -> Parser : Parser("add...")
+activate Parser
+Ui <-- Parser : parser :Parser
+deactivate Parser
+
+Ui --> Parser : parseCommand()
+activate Parser
+
+Parser -> Parser : prepareAdd()
+activate Parser
+
+create AddCommand
+Parser -> AddCommand
+activate AddCommand
+Parser <-- AddCommand
+deactivate AddCommand
+
+Parser --> Parser : :Command
```

Comment:
Maybe it would be good to omit this return line on a self-invocation also? I'm not sure if it conveys any extra information.